### PR TITLE
Fix for (hopefully) all modals not automatically closing

### PR DESCRIPTION
**Pull request by Zingabopp**
Some modals don't close if back out of the view controller, I think this gets them all.

### CODE REVIEW

File: BeatSaberMultiplayer/UI/ViewControllers/CreateRoomScreen/MainRoomCreationViewController.cs [hunk 1]
```diff
@@ -68,10 +68,14 @@ protected override void DidActivate(bool firstActivation, ActivationType activat
...
+            parserParams.EmitEvent("closeAllMPModals");
             if (_presetNameKeyboard != null)
                 _presetNameKeyboard.modalView.Hide(false);
             if (_roomNameKeyboard != null)
+            {
+                Plugin.log.Warn($"Hiding modal keyboard");
                 _roomNameKeyboard.modalKeyboard.modalView.Hide(false);
+            }
             if (_passwordKeyboard != null)
                 _passwordKeyboard.modalKeyboard.modalView.Hide(false);
```
Comment (andruzzzhka) [inline BeatSaberMultiplayer/UI/ViewControllers/CreateRoomScreen/MainRoomCreationViewController.cs]: Is this necessary if we close this keyboard using an event?

File: BeatSaberMultiplayer/UI/ViewControllers/RoomScreen/SongSelectionViewController.cs [hunk 2]
```diff
@@ -74,6 +74,7 @@ protected override void DidActivate(bool firstActivation, ActivationType type)
 
         protected override void DidDeactivate(DeactivationType deactivationType)
         {
+            _parserParams.EmitEvent("closeAllMPModals");
             if (_searchKeyboard != null)
                 _searchKeyboard.modalView.Hide(false);
```
Comment (andruzzzhka) [inline BeatSaberMultiplayer/UI/ViewControllers/RoomScreen/SongSelectionViewController.cs]: This too

File: BeatSaberMultiplayer/UI/ViewControllers/ServerHubScreen/RoomListViewController.cs [hunk 3]
```diff
@@ -44,7 +47,8 @@ protected override void DidActivate(bool firstActivation, ActivationType type)
 
         protected override void DidDeactivate(DeactivationType deactivationType)
         {
-            if(_passwordKeyboard != null)
+            parserParams.EmitEvent("closeAllMPModals");
+            if (_passwordKeyboard != null)
                 _passwordKeyboard.modalView.Hide(false);
```
Comment (andruzzzhka) [inline BeatSaberMultiplayer/UI/ViewControllers/ServerHubScreen/RoomListViewController.cs]: And this

**Comment by Zingabopp** [comment]
Probably not, I feex

**Pull request merged**